source/ecs/entity.d

```diff
@@ -673,38 +623,38 @@ unittest
 @("entity: EntityManager: getOrSet")
 unittest
 {
-	auto em = new EntityManager!size_t();
+	auto em = new EntityManager();
 
 	auto e = em.gen!(Foo)();
 	assertEquals(Foo.init, *em.getOrSet!Foo(e));
 	assertEquals(Foo.init, *em.getOrSet(e, Foo(2, 3)));
 	assertEquals(Bar("str"), *em.getOrSet(e, Bar("str")));
 
-	assertNull(em.getOrSet!Foo(Entity!size_t(0, 12)));
-	assertNull(em.getOrSet!Foo(Entity!size_t(3)));
+	assertNull(em.getOrSet!Foo(Entity(0, 12)));
+	assertNull(em.getOrSet!Foo(Entity(3)));
 }
 
 @safe pure
 @("entity: EntityManager: recycle")
 unittest
 {
 	import std.range : front;
-	auto em = new EntityManager!uint();
+	auto em = new EntityManager();
 
 	auto entity0 = em.gen(); // calls fabricate
 	em.discard(entity0); // discards
-	(() @trusted pure => assertEquals(Entity!uint(0, 1), em.queue.get))(); // batch was incremented
-	assertFalse(Entity!uint(0, 1) == entity0); // entity's batch is not updated
+	(() @trusted pure => assertEquals(Entity(0, 1), em.queue.get))(); // batch was incremented
```

Comment:
the `.queue` here is for internal usage?

Reply:
Yes the `queue` is a `Nullable!Entity` which defines the null value to an entity of id = `Entity.maxid`. Instead of having an array of deleted entities, the entities are swapped with the queue every time they're `disposed` or `revived`. The `queue` has the purpose of tracking the next entity to reuse if not null, aka `Entity(Entity.maxid)`.